### DISCUSSION TITLE
Routing optimization - map interpretation

### DISCUSSION THREAD

**Discussion by lacrimal** [Koenkk/zigbee2mqtt#14303] [Devices]
Hi, 

Can someone explain how connections on (First) are  better than direct (Second) ?
As I understand LQI shows only signal strength from Coordinator to End device.
So only explanation that First is optimal over Second is that communication from End device to nearest router/coordinator is main factor in joining. Am i right ? 
This is probably connected to #2888

First - I understand it as 'optimization' - this is what I see after 12h from connecting devices to Coordinator.

![obraz](https://user-images.githubusercontent.com/22749391/193542568-c5a6b99f-0257-437d-9890-e2bfcd931efb.png)

Second - after forcing reconnect to Coordinator - to show LQI

![obraz](https://user-images.githubusercontent.com/22749391/193543236-93f21391-8922-43b7-9481-2e09af142084.png)

**Comment by JacekKac** [comment]
Mogę tylko pogratulować LQI, ja robię w 3 budynkach i nie osiągam nawet połowy tego :)

**Reply by lacrimal** [reply]
Mam Dongle-E, z definicji ma x2 do LQI. Można go przeprogramować na router. Jeśli masz komunikacyjne problemy to warto sprobować.
Próbowałem to samo osiagnać z CC2652P ale nawet z odblokowanym transmit power 20db zawsze był o 40-50 % słabszy. No i Z2M dev - kuriozalnie - działa stabilniej niż stable.
Tylko to wszystko psu na budę, według teorii to co na rysunku to tylko jakość transmisji od koordynatora.

**Reply by JacekKac** [reply]
co to znaczy dongle-e ?

**Reply by lacrimal** [reply]
Sonoff Dongle-E - nieco nowsza wersja koordynatora do ZigBee. W Z2M jest traktowany jako eksperymentalny, ale z tego co wiem można go łatwo przeprogramować na router ZigBee. A że domyślnie ma duże wzmocnienie w porównaniu do innych dongli - może pomóc w rozszerzaniu zasięgu sieci ZigBee. Podłączasz gdzie w połowie drogi do ładowarki z telefonu na USB i masz lepszy zasięg.

**Reply by JacekKac** [reply]
ah, miałem go tam wetknięty już jako koordynator, nie robi różnicy vs egony ebyte. Mogę jeszcze spróbować, ale mam wrażenie, że na polskie warunki gdzie masz wszystko z betonu to o kant dupy

**Reply by lacrimal** [reply]
Jeśli już miałeś wersję E to nie ma co już próbować, ja specjalnie zamówiłem Sonoff Dongle-E i starszy Dongle-P żeby porownać. Flashować da się bez otwierania - jest interfejs serial na USB. Jest kilka metod, ważne żeby wybrać właściwy firmware reszta w sumie łatwiejsza niż bios w kompie. Pod windowsem jest skrypt pythona do otwarcia portu w trybie botloadera a potem jużsoft do programowania. Jest też opcja zrobienia wszystkiego jednym skryptem. Za drugim razem trwa to minutę :-)
Jak się da to możesz wstawić kilka routerów jeden za drugim (nie koniecznie wyżej wymienionych).

**Reply by JacekKac** [reply]
Nie da się, bo po drodze nie mam prądu nigdzie, a tam gdzie już jest to nic nie łapie. Kombinowałem z anteną, ale to też nic nie daje. Zaatakuję jeszcze z routerem przy kominie, który przechodzi przez strop w piętrze, czasami to pomaga złapać parę kresek zasięgu

**Reply by JacekKac** [reply]
a jeśli nie to rasperry pizero osobny koordynator i PAN i dam przez mqtt do głównego

**Reply by JacekKac** [reply]
nie rozumiem dlaczego zasięg wifi na 2.4 potrafię zlapać pomiędzy piętrami, a na zigbee nie :(

**Reply by lacrimal** [reply]
A jak będziesz zasilać rpi0 ?

**Comment by JacekKac** [comment]
z tym nie ma problemu, bo wifi jest zarowno na górze jak i dole, więc pi postawię gdziekolwiek i tam dam nowy koordynator z nadzieją, że już na piętrze połapie urzadzenia zigbee

**Reply by lacrimal** [reply]
No tak, taki mały mostek może sprawę uratować. Próbowaleś umieszczać router/koordynator z dala od źrodeł prądu i sygnału wifi ? U mnie komputer i telewizor generowały takie zakłócenia że prawie odesłałem dongla. Dopiero po podłączeniu przez kabel 75cm się to całkowicie uspokoiło.

**Reply by JacekKac** [reply]
no stoi pod TV w szafce, oddaliłem od routera o jakiś metr, ale TV jest blisko - oprócz tego podpiąłem antenę z 3m kabla mimo to słabo. Antena obecnie przy TV ale jak stała dalej za ścianką to też nie było lepiej.

**Reply by JacekKac** [reply]
oczywiście oprócz tego też sam dongle na przedłużce 50 cm od pi

**Reply by JacekKac** [reply]
ogólnie rzecz biorąc... jestem już prawie przekonany, że rezygnuję z zigbee. Mam 4 lokalizacje, tylko w jednej działa bez problemów ale to jest 40 m2 i kartongips. Próbowałem już różnych urządzen on budżetowych do droższych i wszędzie efekt jest taki sam. Piętra z betonowym sufitem nie da się pokonać.

**Reply by lacrimal** [reply]
To taki test z przed chwili mam. Wyłączyłem routery o zostawiłem tylko ten mój koordynator z mini termometrem. 12m w linii prostej przez 2 ściany żelbetu. Wifi już mi straciło zasięg w telefonie ale update temperatury poszedł przez ZigBee.

**Reply by JacekKac** [reply]
u mnie : router  w pomieszczeniu mniej więcej centralnie to LQI na jednym piętrze odległości do 5-6 metrów

![image](https://user-images.githubusercontent.com/72306272/193681363-3df23c96-8674-434d-b7f3-52d74a6735d5.png)

**Comment by JacekKac** [comment]
Tak najczęściej wyglądają u mnie przejścia przez piętra 
![image](https://user-images.githubusercontent.com/72306272/193680516-e4c3af21-50ba-4908-8c95-3d84664533ae.png)

**Reply by lacrimal** [reply]
U mnie z wtyczek najlepszy zasięg miała oryginalna  NAS-WR01B

**Comment by JacekKac** [comment]
tutaj jedno piętro odległości do maks 5 metrów (urządzenia Temp 1 do 6(
![image](https://user-images.githubusercontent.com/72306272/193681479-f01c6e4e-5bb7-4543-aefc-9b2ea870bf76.png)

**Reply by JacekKac** [reply]
Temp_1 jest w odległości 80 cm od routera :)

**Reply by lacrimal** [reply]
Kurcze, to zawsze może być problem po dwóch stronach, u mnie te czujniki pastylkowe na cr2032 w odległości 14 metrów mają lqi tak, jak na drugim rysunku - to dwa pokoje obok siebie.
Tyle że nikt mi jeszcze nie odpowiedział w sprawie transmisji od czujnika :-)
Może sproboj tego Dongle-E w wydaniu Sonoff, nawet bez flashowania - u mnie chodzi na tej antence z paczki.

**Reply by JacekKac** [reply]
też na nim robiłem. Sprawdzałem 2 rodzaje czujników sonof znb2 te temperatury i tuyi takie z LCD na baterie normalne. Efekt podobny

**Reply by lacrimal** [reply]
A tak z ciekawości, próbowałeś wyłączyć router? Albo go raczej wynieść gdzieś dalej ?

**Reply by JacekKac** [reply]
router wyłączyć czy repeatery? No kolejna próba to przesunę go w inne miejsce pozbawione urzadzeń elektrycznych

**Reply by lacrimal** [reply]
Router wifi. Wiesz jak odetniesz trochę potencjalnych źródeł interferencji będziesz wiedzieć czy da się z tego coś wycisnąć czy to już jest max.

**Reply by JacekKac** [reply]
Nie, nie próbowałem ale sprawdzałem kanały wifi i zigbee, rozdzieliłem skrajnie.

**Reply by JacekKac** [reply]
Hej, dzięki za tipa z sonoffem E. Faktycznie nie zauważyłem miałem parę plusów i jeden E. Flashowałem go do routera i wpiąłem. Zasięgi rewelacja, wszystko się "widzi". Najlepszy dongle jaki widziałem, a praktycznie używałem wszystkich

**Reply by lacrimal** [reply]
Super, wersja E jest szybka i mocna, co prawda w zigbee2mqtt jeszcze ciągle eksperymentalna ale ZHA wspiera ją już natywnie. A jako router to jest bajka.
